go-controller/pkg/util/net_linux_unit_test.go

```diff
@@ -0,0 +1,951 @@
+			},
+		},
+		{
+			desc: "LinkRoutesAdd() returns NO error when subnets input list is empty",
+		},
+		// TODO: Code may need to be updated to check that link is not-nil to avoid panic
+		/*{
+			desc: "LinkRoutesAdd() panics when link input is nil",
+			inputSubnets: ovntest.MustParseIPNets( "192.168.0.0/24"),
+			inputLink: nil,
+		},*/
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+
+			for _, item := range tc.onRetArgsNetLinkLibOpers {
+				call := mockNetLinkLibOperations.On(item.onCallMethodName)
+				for _, arg := range item.onCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.retArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+
+			for _, item := range tc.onRetArgsLinkIfaceOpers {
+				call := mockLink.On(item.onCallMethodName)
+				for _, arg := range item.onCallMethodArgType {
+					call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+				}
+				for _, ret := range item.retArgList {
+					call.ReturnArguments = append(call.ReturnArguments, ret)
+				}
+				call.Once()
+			}
+
+			err := LinkRoutesAdd(tc.inputLink, tc.inputGwIP, tc.inputSubnets)
+			t.Log(err)
+			if tc.errExp {
+				assert.Error(t, err)
+			}
+			mockNetLinkLibOperations.AssertExpectations(t)
+			mockLink.AssertExpectations(t)
+		})
+	}
+}
+
+func TestLinkRouteExists(t *testing.T) {
+	mockNetLinkLibOperations := new(mocks.NetLinkLibOperations)
+	// below is defined in net_linux.go
+	netLinkOperations = mockNetLinkLibOperations
+
+	tests := []struct {
+		desc                     string
+		inputLink                netlink.Link
+		inputGwIP                net.IP
+		inputSubnet              *net.IPNet
+		errExp                   bool
+		outBoolFlag              bool
+		onRetArgsNetLinkLibOpers []onCallReturnArgs
+	}{
+		// TODO: The `link` argument to the function LinkRouteExists is never used in the funciton body and could be removed?
+		{
+			desc:        "tests code path when RouteListFiltered() returns error",
+			inputGwIP:   ovntest.MustParseIP("192.168.0.1"),
+			inputSubnet: ovntest.MustParseIPNet("192.168.0.0/24"),
+			errExp:      true,
+			outBoolFlag: false,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"RouteListFiltered", []string{"int", "*netlink.Route", "uint64"}, []interface{}{[]netlink.Route{}, fmt.Errorf("mock error")}},
+			},
+		},
+		{
+			desc:        "tests code path when RouteListFiltered() returns empty routes list",
+			inputGwIP:   ovntest.MustParseIP("192.168.0.1"),
+			inputSubnet: ovntest.MustParseIPNet("192.168.0.0/24"),
+			outBoolFlag: false,
+			onRetArgsNetLinkLibOpers: []onCallReturnArgs{
+				{"RouteListFiltered", []string{"int", "*netlink.Route", "uint64"}, []interface{}{[]netlink.Route{}, nil}},
+			},
+		},
+		// TODO: Might be a good idea to check for empty Gateway IP in the function body and return early
```

Comment:
it's caller's responsibility to ensure they are non-nil.

Reply:
ok